### ISSUE TITLE
:ajax not sending parameter

### ISSUE THREAD

**Issue by cod3rshotout** [sebastienheyd/boilerplate#54]
I'm trying to use the `:ajax` feature, but for some reason I'm not able to make it working. This is my datatable:

    <x-boilerplate::datatable name="medicalrecordsriskfield" :ajax="['riskarea_id' => $riskarea->id]" />

within the Datatable `datasource` method I have:

        dd(request()->post('riskarea_id'));

and it returns `null`.

**UPDATE**

The error is related to this line:

    window.{{ $id }}_ajax = {!! json_encode($ajax) !!}

which actually does nothing since the ajax data must be passed in the ajax block:

```
ajax: {
    url: '{!! route('boilerplate.datatables', $datatable->slug, false) !!}',
    type: 'post',
    data: $.fn.dataTable.parseDatatableFilters,
    complete: () => {
        $('#{{ $id }} [name=dt-check-all]').prop('checked', false)
    }
},
```

In particular within the data block, but I'm not sure the purpose of this `$.fn.dataTable.parseDatatableFilters`, so I will wait for a fix

**Comment by sebastienheyd** [comment]
Do you have the latest version of Boilerplate ?

**Comment by cod3rshotout** [comment]
> Do you have the latest version of Boilerplate and updated the assets ?
> 
> ```
> php artisan vendor:publish --tag=laravel-assets --force
> ```
> 
> The empty posted values with ajax have been fixed in the version [7.14.0](https://github.com/sebastienheyd/boilerplate/releases/tag/7.14.0)

I mean the Ajax parameter on the Datatable that is not sending data, not the filters. If you add :ajax on the Datatable and then try to get the value from request, this is always null 'cause the Datatable isn't sending it

**Comment by sebastienheyd** [comment]
Can you check these points :

1. When you add `name="users" :ajax="['role' => 'admin55555']"` the array that will be added to the ajax call must appear in the DOM as a json. Try to see if you can find the json in the DOM by searching the variable `window.dt_users_ajax` in the page HTML source.

![Capture d’écran du 2022-03-03 10-01-14](https://user-images.githubusercontent.com/5470423/156531901-73c06f0d-21f2-4f5d-ae10-36d33aa4e08f.png)

2. The JSON is sent with the filters by extending the filters array, see [method parseDatatableFilters in datatables.js - ln.122](https://github.com/sebastienheyd/boilerplate/blob/master/src/resources/assets/js/datatables.js#L122). Check if role is correctly posted with the filters in the network monitor. 

![Capture d’écran du 2022-03-03 10-20-29](https://user-images.githubusercontent.com/5470423/156534790-b0490237-eb5a-478c-af7e-2bdcb258d486.png)

![Capture d’écran du 2022-03-03 10-00-36](https://user-images.githubusercontent.com/5470423/156531956-5c7e3370-acb0-4754-b0e8-28ca427a6470.png)

If not, this is a problem with the asset, try to find `$.extend(!0,e,window[n+"_ajax"])` in `public/assets/vendor/boilerplate/plugins/datatables/datatables.min.js`

3. If everything is ok, you must have access to the posted value

![Capture d’écran du 2022-03-03 10-28-28](https://user-images.githubusercontent.com/5470423/156536212-e1a52e76-51ef-4603-92a6-9f12f665d60e.png)

![Capture d’écran du 2022-03-03 10-07-21](https://user-images.githubusercontent.com/5470423/156536262-89ede9e4-3df6-40fa-8e41-3434deee6b93.png)
